### ISSUE TITLE
Multicolor and Background controls - Problem with Restore the autosave

### ISSUE THREAD

**Issue by Radzio1615** [themeum/kirki#1942]
### Issue description:
I noticed that when I use '**multicolor**' or '**background**' controls and when I refresh customizer without saving I don't get notification "_There is a more recent autosave of yout changes than the one you are previewing. Restore the autosave_". Is there a way to fix this? Does anyone also have this problem?

### Version used:
Kirki 3.0.33

### Using theme_mods or options?
theme_mods

**Comment by aristath** [comment]
I don't quite get the issue here...
Could you please elaborate?
Do you change a value in those controls and on refresh they're not there? What is it that other controls do and these don't? Can can I replicate this?

**Comment by Radzio1615** [comment]
When I don't use in my project "multicolor" and "background" controls, notification about possibility to restore autosave properly appear.
![properly appear](https://i.imgur.com/HZPoiwp.png)
If I add one or both of this controls notification disappear and "Publish" button are always active (label on the button don't change to "Published" but changes are saved after I click).

Code looks like this:
```
Kirki::add_field( 'theme_config_id', array(
	'type'        => 'multicolor',
	'settings'    => 'button_bg,
	'label'       => esc_attr__( 'Button background color' ),
	'section'     => 'section_ui',
	'priority'    => 10,
	'choices'     => array(
		'hover'    => esc_attr__( 'Hover' ),
		'active'   => esc_attr__( 'Active' ),
	),
	'default'     => array(
		'hover'    => '#000',
		'active'   => '',
	),
	'output'      => array(
		array(
			'element'	=> 'button:hover',
			'property'	=> 'background-color',
			'choice'   => 'hover',
		),
		array(
			'element'	=> 'button:active',
			'property'	=> 'background-color',
			'choice'   => 'active',
		),
	),
) );
}

```

**Comment by seventhqueen** [comment]
I can confirm that if you have a background field added, when opening the customizer the Publish button gets activated and it somehow will think that the field value is changed but it is not. 

You can check it by adding this snippet of code and it will say that the fields are changed, even at the first open of the customizer: 
```
if ( is_customize_preview() ) {
	if ( isset( $_POST['customized'] ) && ! empty( $_POST['customized'] ) && $_POST['customized'] != '{}' ) {
		var_dump($_POST['customized']);exit;
	}
}
```

**Comment by seventhqueen** [comment]
Hello, thanks for looking into this, Maybe an option to disable the color control from the background field

**Comment by aristath** [comment]
Just an update here:
A few months ago we had this problem with normal color fields as well, however I started rewriting controls in preparation for v3.1 and this was fixed for plain color controls when that was done.

Fixing this issue requires rewriting the typography, background, multicolor & palette controls.
It will be done, but it depends on first rewriting some more basic fields the ones that are part of these more complex controls.
In v3.0.34 (develop branch) I refactored the `image` field which is part of the background control.
color, text, select & radio have already been migrated to the new structure in previous releases and have had months of testing so we know they're ok.

I know this is an annoying issue, but the good news is that it doesn't actually break any sites, so I'd like to first give the `image` control at least a month of testing on live sites after v3.0.34 is released. Once we verify everything works fine with that and we fix any and all bugs that may come up with `image` controls, then I can start rebuilding the more complex fields - like background.
In other words, please be a little patient with this one... I want it done right instead of just adding patches on top of patches on top of an ill-conceived implementation.

Until v3.0.34 is released, you can help by testing the develop branch and reporting any issues you may find with the refactored image controls.

**Comment by mapsteps** [comment]
Hello and thank you for reporting this issue.

We are closing most open GitHub issues with the release of Kirki 4 to start fresh with this major release.

If you are still running into issues after you have tested your theme with Kirki 4, please let us know so we can reopen the issue.

Please note that Kirki 3 is no longer actively supported. Moving forward, we will only provide support the very latest version of Kirki.

Thank you for your understanding.

Best,
David
